### DISCUSSION TITLE
RaisimGymTorch example with Atlas

### DISCUSSION THREAD

**Discussion by tseyde** [raisimTech/raisimLib#57] [Q&A]
I am looking into setting up a RaisimGymTorch example with Atlas following the provided ANYmal example.
May I ask whether this has been tried before? Would you have some advice on parameters that work well?
I have selected some values for the initial configuration and PD gains, but I assume that quality of these values strongly correlates with quality of the resulting learning.

Thanks a lot for providing this amazing simulator!

**Comment by jhwangbo** [comment]
You are right. PD gains will make a huge impact on the learning performance. I haven't tried with the atlas model. For the PD gains, I guess you can check their mass matrix and use the diagonal terms to tune (e.g., get the same acceleration for all joints at the max torque). Maybe you have to tune further from there due to the ground interaction.
Note that the provided PD controller is a stable PD controller. It will be much more stable than a discrete-time PD controller so the gains are not so critical for stability.
You have to be careful with the contact geometry. I noticed that the common Atlas urdf (probably from DARPA) are not even simulatable due to collision locking. I guess they just ignored the internal collisions when they set up the GAZEBO environment. We provide the atlas model with cleaned up collision geoms.

**Reply by tseyde** [reply]
Thank you so much for your in-depth feedback and advice on how to tune the gains!
I will try this more structured approach to setting the configuration & gains. :)
